ignite/metrics/ssim.py

```diff
@@ -158,8 +159,21 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
             )
 
         channel = y_pred.size(1)
-        if len(self._kernel.shape) < 4:
-            self._kernel = self._kernel.expand(channel, 1, -1, -1).to(device=y_pred.device)
+
+        self._kernel = self._kernel.expand(channel, 1, -1, -1)
```

Comment:
I have some doubts about expanding all the times. I'm not sure what would be the best way of doing that especially taking into account that `channel` can potentially vary across the dataset.

Reply:
My last commit passed the tests and solve this issue